hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusCommandContext.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.bus;
+
+import javax.jms.ConnectionFactory;
+
+import org.hawkular.bus.common.Endpoint;
+import org.hawkular.cmdgw.command.ws.ConnectedWsSessions;
+
+/**
+ * Context information that may be useful during an execution of a {@link BusCommand}.
+ *
+ * @see BusCommand#execute(org.hawkular.bus.common.BasicMessageWithExtraData, BusCommandContext)
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class BusCommandContext {
+
+    private final ConnectedWsSessions connectedFeeds;
+    private final ConnectedWsSessions connectedUIClients;
+    private final ConnectionFactory connectionFactory;
+    private final Endpoint queue;
+
+    public BusCommandContext(Endpoint queue, ConnectionFactory connectionFactory,
+            ConnectedWsSessions connectedUIClients, ConnectedWsSessions connectedFeeds) {
+        super();
+        this.queue = queue;
+        this.connectionFactory = connectionFactory;
+        this.connectedUIClients = connectedUIClients;
+        this.connectedFeeds = connectedFeeds;
+    }
+
+    public ConnectedWsSessions getConnectedFeeds() {
+        return connectedFeeds;
+    }
+
+    public ConnectedWsSessions getConnectedUIClients() {
+        return connectedUIClients;
+    }
+
+    public ConnectionFactory getConnectionFactory() {
+        return connectionFactory;
+    }
+
+    /**
+     * @return the queue the present request came from
+     */
+    public Endpoint getQueue() {
```

Comment:
Endpoints may not be queues in the future. I have a feeling we may want to use Topics in the future. I would just name this "endpoint" and not specifically "queue" -- change the javadoc and method name to indicate "endpoint"